src/app/public/modules/demo-page/demo-page-dom-adapter.service.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Injectable,
+  OnDestroy
+} from '@angular/core';
+
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  SkyDocsAnchorLink
+} from '../type-definitions/anchor-link';
+
+/**
+ * @internal
+ */
+@Injectable()
+export class SkyDocsDemoPageDomAdapterService implements OnDestroy {
```

Comment:
No new logic, here. Just moving the DOM-related code from the demo-page component to a separate adapter service.